pipeline/00-ingest.R

```diff
@@ -270,6 +271,25 @@ training_data_clean <- training_data_w_hie %>%
   # This will remove any categories not stored in the dictionary and convert
   # them to NA (useful since there are a lot of misrecorded variables)
   ccao::vars_recode(cols = starts_with("char_"), type = "code") %>%
+  # Recode the number of apartments from its numeric code to its actual number
+  # of units. Additionally, ensure non-multi-family PINs always have NONE apts
+  ccao::vars_recode(
+    cols = all_of("char_apts"),
+    type = "short",
+    as_factor = FALSE
+  ) %>%
+  mutate(
+    char_apts = case_when(
+      char_class %in% c("211", "212") & !is.na(char_apts) ~ char_apts,
+      char_class %in% c("211", "212") & is.na(char_apts) ~ "UNKNOWN",
+      TRUE ~ "NONE"
+    ),
+    char_apts = factor(
+      char_apts,
+      levels = c("TWO", "THREE", "FOUR", "FIVE", "SIX", "UNKNOWN", "NONE")
+    ),
+    char_ncu = ifelse(char_class == "212" & !is.na(char_ncu), char_ncu, 0)
+  ) %>%
```

Comment:
Lots of properties that _aren't_ multi-family still have a `char_apts` value that isn't `NONE`. This is likely from conversions and such where the class is changed but the `APTS` variable isn't. We want `char_apts` and `char_ncu` set only for their respective classes, so we force that here.